lib/core-index-stream.js

```diff
@@ -105,6 +111,11 @@ class CoreIndexStream extends Readable {
     this.#inProgressBitfield?.set(index, false)
   }
 
+  async unlink() {
+    this.#storage ??= await this.#createStorage()
+    await unlinkStorage(this.#storage)
```

Comment:
i'm assuming there's an edge case that this accounts for but just in case: would a no-op (or throw) be possible here if the storage isn't initialized? e.g. `if (!this.#storage) return`

this seems to be performing operations that cancel each other out if the storage isn't already initialized, but maybe it's much more complex than that

Reply:
There might be cases where the storage has been created on disk but hasn't been initialized in memory. For example:

1. I start the app, index a bunch of stuff, and then close it.
2. Later, I start the app and immediately unlink. `this.#storage` won't be created. We want to create it so we can unlink it.

My understanding is that creating the storage instance doesn't actually touch any files on disk. For example:

```javascript
const RandomAccessFile = require("random-access-file")
const fs = require("node:fs")

new RandomAccessFile("my-file.txt")

fs.existsSync("my-file.txt")
// => false
```

Maybe it's worth adding an explainer comment about this?

Reply:
ah okay that makes more sense! I was under the impression that this class fully manages the storage on disk but if it's being done somewhere else, your reasoning is clearer. thanks!